### PR TITLE
Add `@FileExists` validation

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Fail-fast in case that the file referenced in the configuration does not exist.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by Praveen2112 — APPROVED** [review]

**Review by ebyhr — APPROVED** [review]
Could you fix `TestS3SecurityMappingConfig` too?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfig.java [hunk 1]
```diff
@@ -30,7 +31,7 @@
     private Duration refreshPeriod;
     private String colonReplacement;
 
-    public Optional<String> getConfigFilePath()
+    public Optional<@FileExists String> getConfigFilePath()
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfig.java:34]: The path might be http(s). Does `@FileExists` annotation work even for such protocols?
https://github.com/airlift/airlift/blob/master/configuration/src/main/java/io/airlift/configuration/validation/FileExistsValidator.java

Comment (Praveen2112) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfig.java:34]: I think it won't work for http url. Can we revert this change for this class.

Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfig.java]: Good catch. Thank you for spotting this one.
Are there multiple situations like this one which would warrant a `@ResourceExists` annotation / extension of the `FileExistsValidator`?

Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfig.java]: we won't do any HTTP calls during config validation

**Comment by findinpath** [comment]
CI hit https://github.com/trinodb/trino/issues/12535

**Pull request merged**